liberty-ls/src/main/java/io/openliberty/tools/langserver/utils/Messages.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+* Copyright (c) 2022 IBM Corporation and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0
+*******************************************************************************/
+package io.openliberty.tools.langserver.utils;
+
+import java.util.Locale;
+import java.util.ResourceBundle;
+import java.util.regex.Pattern;
+
+public class Messages {
+    private static ResourceBundle serverenvMessages = null;
+    private static ResourceBundle bootstrapMessages = null;
+
+    // Matches server.env key formats: all uppercase, possibly with underscores and additional uppercase words
+    private static Pattern serverEnvKeyFormat = Pattern.compile("([A-Z]+)(_[A-Z]+)*");
+
+    private static synchronized void initializeBundles() {
+        Locale locale = Locale.getDefault(); //TODO: properly set/get locale
+        serverenvMessages = ResourceBundle.getBundle("ServerEnv", locale);
+        bootstrapMessages = ResourceBundle.getBundle("BootstrapProperties", locale);
+    }
+
+    public static String getPropDescription(String key) {
+        if (serverenvMessages == null) {
+            initializeBundles();
+        }
+        String message = null;
+        if (serverEnvKeyFormat.matcher(key).matches()) { // server env
+            message = serverenvMessages.getString(key);
+        } else if (key.contains(".")) { // bootstrap property
+            message = bootstrapMessages.getString(key);
+            // 2nd lookup for properties that have a server.env equivalent
+            if (serverEnvKeyFormat.matcher(message).matches()) {
```

Comment:
Can you explain this logic a bit? If the description is not in bootstrapMessages, what does it return? How is that going to match something in serverEnvKeyFormat regex?

Reply:
Currently, some of the entries in `BootstrapProperties.properties` are like this: `com.ibm.ws.logging.log.directory=LOG_DIR`, and `LOG_DIR` is defined in the ServerEnv.properties
If the value of the bootstrap matches the format for a server.env key, then it does a lookup on the ServerEnv properties to get its description.

For either server.env or bootstrap.properties, if the key is not found, then no hover appears. It currently throws a MissingResourceException for not being able to find the key, but doesn't break anything. May add in some exception handling for it.

Reply:
Ok, got it. Doesn't seem like throwing an exception when a key is not found makes sense though. There can definitely be keys that our LS will not recognize but would be perfectly valid.

Reply:
The current Exception doesn't cause any problems with the language server continuing to run, it just provides some information on what key was not found.

I may add a try/catch for it to clean up the console output.